jet_tracking/jet_control.py

```diff
@@ -160,49 +175,89 @@ def calibrate_inline(injector, camera, params, *, settle_time=1.0,
     burst_imagess : int, optional
         Number of burst images to average from the camera
     '''
+
     injector_axis = injector.coarseZ
+    # collect images and motor positions needed for calibration
     positions, imgs = get_calibration_images(injector_axis, camera,
                                              settle_time=settle_time,
                                              burst_images=burst_images)
 
+    # cam_roll: rotation of camera about z axis in radians
+    # pxsize: size of pixel in mm
     cam_roll, pxsize = cam_utils.get_cam_roll_pxsize(imgs, positions)
     params.pxsize.put(pxsize)
     params.cam_roll.put(cam_roll)
 
+    # beam_x_px: x-coordinate of x-ray beam in camera image in pixels
     beam_x_px = params.beam_x_px.get()
+    # beam_y_px: y-coordinate of x-ray beam in camera image in pixels
     beam_y_px = params.beam_y_px.get()
+    # cam_x: x-coordinate of camera position in mm
+    # cam_y: y-coordinate of camera position in mm
     cam_x, cam_y = cam_utils.get_cam_coords(beam_x_px, beam_y_px,
                                             cam_roll=cam_roll, pxsize=pxsize)
     params.cam_x.put(cam_x)
     params.cam_y.put(cam_y)
 
-    # find jet in camera ROI
-    ROI_image = cam_utils.get_burst_avg(burst_images, camera.ROI_image)
-    rho, theta = cam_utils.jet_detect(ROI_image)
-
-    jet_roll = cam_utils.get_jet_roll(theta, cam_roll=cam_roll)
+    # jet_roll: rotation of sample jet about z axis in radians
+    jet_roll = cam_utils.get_jet_roll(params.theta.get(), cam_roll=cam_roll)
     params.jet_roll.put(jet_roll)
-    return dict(jet_roll=jet_roll, rho=rho, theta=theta, pxsize=pxsize,
+    return dict(jet_roll=jet_roll, pxsize=pxsize,
                 cam_roll=cam_roll)
 
 
-def calibrate(injector, camera, params, *, offaxis=False, settle_time=0.1):
+def calibrate(injector, camera, cspad, wave8, gas_det, params, *, offaxis=False, settle_time=0.1):
     '''
-    Calibrate the camera
+    Calibrate the camera and CSPAD and determine parameters needed for
+    jet tracking
+    NEED TO CHECK offaxis calculation sign
+
+    First set the ROI of the camera to show the proper jet and illumination.
+
+    Determines the mean, standard deviation, radius, intensity, jet position and
+    tilt, pixel size, camera position and tilt
+
+    Params determined if onaxis camera used: mean, std, radius, intensity, pxsize,
+    camX, camY, cam_roll, jet_roll
+
+    Params determined if offaxis camera used: mean, std, radius, intensity, pxsize,
+    camY, camZ, cam_pitch, jet_pitch
 
     Parameters
```

Comment:
More description of what these are is desired

Reply:
My brain has so far rejected any interactions with these functions that convert the image analysis result into something useable for jet tracking. At some point Meng had agreed to look into that part of the code. Would there be a way to include a picture somewhere in the docs to show what this is supposed to do?